src/server.tsx

```diff
@@ -103,14 +114,68 @@ const buildEpic = async (
     return asResponse(getEpic(props), testTracking, result.debug);
 };
 
+const buildEpicData = async (
+    pageTracking: EpicPageTracking,
+    targeting: EpicTargeting,
+    params: Params,
+): Promise<DataResponse> => {
+    const configuredTests = await fetchConfiguredEpicTestsCached();
+    const hardcodedTests = await getAllHardcodedTests();
+    const tests = [...configuredTests.tests, ...hardcodedTests];
+
+    let result: Result;
+
+    if (params.force) {
+        const test = tests.find(test => test.name === params.force?.testName);
+        const variant = test?.variants.find(v => v.name === params.force?.variantName);
+        result = test && variant ? { result: { test, variant } } : {};
+    } else {
+        result = findTestAndVariant(tests, targeting, params.debug);
+    }
+
+    logTargeting(
+        `Renders Epic ${result ? 'true' : 'false'} for targeting: ${JSON.stringify(targeting)}`,
+    );
+
+    if (!result.result) {
+        return { data: undefined, debug: result.debug };
+    }
+
+    const { test, variant } = result.result;
+
+    const testTracking: EpicTestTracking = {
+        abTestName: test.name,
+        abTestVariant: variant.name,
+        campaignCode: buildCampaignCode(test, variant),
+        campaignId: `epic_${test.campaignId || test.name}`,
+    };
+
+    const props = {
+        variant,
+        tracking: { ...pageTracking, ...testTracking },
+        numArticles: getArticleViewCountForWeeks(targeting.weeklyArticleHistory),
+        countryCode: targeting.countryCode,
+    };
+
+    return {
+        data: {
+            variant,
+            meta: testTracking,
+            moduleUrl: 'https://contributions.theguardian.com/epic.js',
```

Comment:
Another optimisation, but I think there's argumenthere that this should be environment aware. Perhaps serving only the module name and let the platform build the URL by looking at some environment config? This would be consistent with how platforms talk to our APIs.